WordPress/gutenberg
# Add group block

# Now renamed to "Section" (by popular demand)

-------

Closes #4900

## Description
Adds a basic version of a new Container block. This block supports:
- Align wide & full
- Anchor links
- Background color
- ~The ability to remove the default paddin~

Not included:
- Text color—this could be added, but we thought we'd keep the initial PR small.
- Background images—see the discussion on the issue
- Inner block alignment—again, something that can be addressed in a later PR
- Block appender from the mockups—`InnerBlocks` doesn't support this out of the box, so it will need some exploration in [a separate PR.](https://github.com/WordPress/gutenberg/pull/14241)


## Todo
- [x] Should padding match `paragraph`?
- [x] Confirm the 1px overflow is gone (best tested in FF)?
- [x] Rebase branch
- [x] Should the Block be `reusable`?
- [x] Should we remove the feature flag?
- [x] Decide on naming `Section` vs `Container` - then update all references
- [x] Remove default background in favour of merging changes from https://github.com/WordPress/gutenberg/pull/14241
- [x] Solve the issue of the [movement handles being invisible when the Block is full width](https://github.com/WordPress/gutenberg/pull/13964#issuecomment-471625001).
- [x] Validation error (on reload) when _custom_ background color is applied 
- [x] Feature flag the section block.

## What to test

If you'd like to test this Block the main features to focus on are:

1. Ability to nest other Blocks within the Section Block.
2. Ability to align the Block itself _and_ all child Blocks (only those which support the alignment settings).

For #2 you can utilise [@getdave's Plugin](https://github.com/getdave/gutenberg-section-block-alignment-tester) which will auto-generate a series of `Section` Blocks for you in various alignment combinations. 

A good first test scenario might be:

1. Add some standard Blocks
2. Add a Section Block - how did does that process feel?
3. Add Blocks within the Section - does this feel intuative?
4. Experiment with some different alignment settings on the Section itself - does it respond as expected?
5. Try changing the alignment settings of some child Blocks which support alignmnet (eg: `cover`, `columns`, `image`...etc). Do these settings respond as expected? How about when you change the Section Block alignment settings at the same time?
6. Add / remove background colors - is this a seemless process?

As a second pass find some [common web layouts](https://wordpress.org/themes/) and try and recreate these using the Section Block. 

## How has this been tested?
1. Add a Section block
2. Try adding inner blocks (this container block supports all blocks)
3. Try adding background colors and using different alignments
4. Preview the post

Expected result:
The ~container~ Section block should appear as expected

## Screenshots 
<img width="906" alt="screen shot 2019-02-20 at 5 21 27 pm" src="https://user-images.githubusercontent.com/677833/53080625-0a5dc600-3534-11e9-9464-484a2445b5f0.png">

## Types of changes


New feature (non-breaking change which adds functionality)




## Checklist:
